doc/developers/contributing.rst

```diff
@@ -169,10 +169,10 @@ Contributing code
 
 .. note::
 
-  To avoid duplicating work, it is highly advised that you contact the
-  developers on the mailing list before starting work on a non-trivial feature.
-
-  https://mail.python.org/mailman/listinfo/scikit-learn
+  To avoid duplicating work, it is highly advised that you search through the
+  issue tracker and the PR list. If you want to work on a non-trivial feature,
```

Comment:
What do you think about adding that when in doubt about duplicated work it is always wise to open an issue first (even for a trivial feature) because the core developers might know that the suggested enhancement/fix is already being addressed (or already done in master)?

Reply:
Maybe keep the note short? A duplicate PR is similar to a duplicate issue IMO.

Reply:
> A duplicate PR is similar to a duplicate issue IMO.

Except if the PR is opened with substantial work already done. 

What about just adding to the last sentence: "When in doubt about duplicated work or if you want to work on a non-trivial feature, it's recommended to first open an issue..."?

Reply:
Or simply, "If in doubt, or if you want to .... "